homeassistant/components/fitbit/sensor.py

```diff
@@ -731,10 +751,12 @@ def __init__(
         super().__init__(coordinator)
         self.entity_description = description
         self.device = device
-        self._attr_unique_id = f"{user_profile_id}_{description.key}"
-        if device is not None:
-            self._attr_name = f"{device.device_version} Battery"
-            self._attr_unique_id = f"{self._attr_unique_id}_{device.id}"
+        self._attr_unique_id = f"{user_profile_id}_{description.key}_{device.id}"
```

Comment:
We're changing unique IDs, do we have to migrate them?

Reply:
The intent is to leave these identical and remove the convoluted logic for creating it given Device is always present. (see if you agree it's not really changing)  Its here to simplify reading the difference between the device info identifier and the entity unique id and make it comparable to the other battery sensor, though I can move this to a separate PR if preferred.

Reply:
Was the type annotation for `device` incorrect? It wasn't optional.

Reply:
There used to be a single sensor class that combined the battery sensors that accepted a `device` and the other sensors where it was not an argument.  I split the sensors into two separate entities and made `device` required for the battery sensor, but i think i left this check.